theano/tensor/extra_ops.py

```diff
@@ -1116,34 +1116,64 @@ class Unique(theano.Op):
     >>> import theano
 
     >>> x = theano.tensor.vector()
-    >>> f = theano.function([x], Unique(True, True, False)(x))
+    >>> f = theano.function([x], Unique(True, True, False, None)(x))
     >>> f([1, 2., 3, 4, 3, 2, 1.])
     [array([ 1.,  2.,  3.,  4.]), array([0, 1, 2, 3]), array([0, 1, 2, 3, 2, 1, 0])]
 
     >>> y = theano.tensor.matrix()
-    >>> g = theano.function([y], Unique(True, True, False)(y))
+    >>> g = theano.function([y], Unique(True, True, False, None)(y))
     >>> g([[1, 1, 1.0], (2, 3, 3.0)])
     [array([ 1.,  2.,  3.]), array([0, 3, 4]), array([0, 0, 0, 1, 2, 2])]
 
+    >>> z = theano.tensor.matrix()
+    >>> h1 = theano.function([y], Unique(True, True, True, 0)(z))
+    >>> h1([[[0, 1, 1], [0, 1, 1]], [[0, 1, 1], [0, 1, 1]], [[0, 0, 0], [0, 1, 1]], [[3, 2, 0], [0, 1, 1]]])
+    [array([[[0., 0., 0.], [0., 1., 1.]], [[0., 1., 1.], [0., 1., 1.]], [[3., 2., 0.], [0., 1., 1.]]]), array([2, 0, 3]), array([1, 1, 0, 2]), array([1, 2, 1])]
+
+    >>> h2 = theano.function([y], Unique(True, True, True, None)(z))
+    >>> h2([[[0, 1, 1], [0, 1, 1]], [[0, 1, 1], [0, 1, 1]], [[0, 0, 0], [0, 1, 1]], [[3, 2, 0], [0, 1, 1]]])
+    [array([0., 1., 2., 3.]), array([ 0,  1, 19, 18]), array([0, 1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 3, 2, 0, 0, 1, 1]), array([10, 12,  1,  1])]
+
     """
 
-    __props__ = ("return_index", "return_inverse", "return_counts")
+    __props__ = ("return_index", "return_inverse", "return_counts",
+                 "axis")
 
     def __init__(self, return_index=False, return_inverse=False,
-                 return_counts=False):
+                 return_counts=False, axis=None):
         self.return_index = return_index
         self.return_inverse = return_inverse
         self.return_counts = return_counts
+        self.axis = axis
         numpy_ver = [int(n) for n in np.__version__.split('.')[:2]]
         if self.return_counts and bool(numpy_ver < [1, 9]):
             raise RuntimeError(
                 "Numpy version = " + np.__version__ +
                 ". Option 'return_counts=True' works starting"
                 " from version 1.9.0.")
```

Comment:
This check can probably be removed since we already require NumPy 1.9+

Reply:
I'll remove it